diff -urN elemental-0.80.orig/tests/core/DifferentGrids.cpp elemental-0.80/tests/core/DifferentGrids.cpp
--- elemental-0.80.orig/tests/core/DifferentGrids.cpp	2013-06-14 04:06:00.000000000 +0900
+++ elemental-0.80/tests/core/DifferentGrids.cpp	2013-09-10 16:32:58.000000000 +0900
@@ -40,7 +40,7 @@
         mpi::GroupIncl( group, sqrtRanks.size(), &sqrtRanks[0], sqrtGroup );
 
         const Grid grid( comm );
-        const Grid sqrtGrid( comm, sqrtGroup );
+        const Grid sqrtGrid( comm, sqrtGroup, grid.Height() );
 
         DistMatrix<double> A(grid), ASqrt(sqrtGrid);
 
