--- config/BuildSystem/config/setCompilers.py.orig	2013-06-27 19:03:10.000000000 +0900
+++ config/BuildSystem/config/setCompilers.py	2013-06-06 22:38:49.000000000 +0900
@@ -979,7 +979,7 @@
       self.pushLanguage(language)
       #different compilers are sensitive to the order of testing these flags. So separete out GCC test.
       if config.setCompilers.Configure.isGNU(self.getCompiler()): testFlags = ['-fPIC']
-      else: testFlags = ['-PIC', '-fPIC', '-KPIC','-qpic']
+      else: testFlags = ['-KPIC', '-fPIC', '-PIC','-qpic']
       for testFlag in testFlags:
         try:
           self.framework.logPrint('Trying '+language+' compiler flag '+testFlag)
