trilinos-11.8.1-Sourceだけに発見: SIERRA
diff -ur --ignore-all-space trilinos-11.8.1-Source.orig/packages/ThreadPool/src/TPI.c trilinos-11.8.1-Source/packages/ThreadPool/src/TPI.c
--- trilinos-11.8.1-Source.orig/packages/ThreadPool/src/TPI.c	2014-04-30 01:14:17.000000000 +0900
+++ trilinos-11.8.1-Source/packages/ThreadPool/src/TPI.c	2014-08-03 05:09:53.000000000 +0900
@@ -68,7 +68,7 @@
 #define THREADPOOL_CONFIG "PTHREAD SCHED_YIELD"
 
 #elif	defined( __linux__ ) && \
-	defined( __GNUC__ ) && ( 4 <= __GNUC__ )
+        defined( __GNUC__ ) && ( 4 <= __GNUC__ ) && !defined(__fcc_version)
 
 #define THREADPOOL_CONFIG "PTHREAD SCHED_YIELD ATOMIC_SYNC"
 
trilinos-11.8.1-Source/packages/ThreadPool/srcだけに発見: TPI.c~
diff -ur --ignore-all-space trilinos-11.8.1-Source.orig/packages/ifpack/src/Ifpack_OverlappingPartitioner.cpp trilinos-11.8.1-Source/packages/ifpack/src/Ifpack_OverlappingPartitioner.cpp
--- trilinos-11.8.1-Source.orig/packages/ifpack/src/Ifpack_OverlappingPartitioner.cpp	2014-04-30 01:14:07.000000000 +0900
+++ trilinos-11.8.1-Source/packages/ifpack/src/Ifpack_OverlappingPartitioner.cpp	2014-08-03 04:41:35.000000000 +0900
@@ -224,7 +224,7 @@
 
 	  // has this column already been inserted?
 	  std::vector<int>::iterator
-	    where = find(tmp[part].begin(), tmp[part].end(), col);
+	    where = std::find(tmp[part].begin(), tmp[part].end(), col);
 
 	  if (where == tmp[part].end()) {
 	    tmp[part].push_back(col);
trilinos-11.8.1-Source/packages/ifpack/srcだけに発見: Ifpack_OverlappingPartitioner.cpp~
diff -ur --ignore-all-space trilinos-11.8.1-Source.orig/packages/ifpack/src/Ifpack_OverlappingRowMatrix.cpp trilinos-11.8.1-Source/packages/ifpack/src/Ifpack_OverlappingRowMatrix.cpp
--- trilinos-11.8.1-Source.orig/packages/ifpack/src/Ifpack_OverlappingRowMatrix.cpp	2014-04-30 01:14:07.000000000 +0900
+++ trilinos-11.8.1-Source/packages/ifpack/src/Ifpack_OverlappingRowMatrix.cpp	2014-08-03 04:43:54.000000000 +0900
@@ -106,7 +106,7 @@
       int_type GID = (int_type) ColMap->GID64(i); 
       if (A().RowMatrixRowMap().LID(GID) == -1) { 
         typename std::vector<int_type>::iterator pos 
-          = find(ExtElements.begin(),ExtElements.end(),GID); 
+          = std::find(ExtElements.begin(),ExtElements.end(),GID);
         if (pos == ExtElements.end()) { 
           ExtElements.push_back(GID);
           list[count] = GID; 
trilinos-11.8.1-Source/packages/ifpack/srcだけに発見: Ifpack_OverlappingRowMatrix.cpp~
diff -ur --ignore-all-space trilinos-11.8.1-Source.orig/packages/ml/src/MatrixFree/ml_MatrixFreePreconditioner.cpp trilinos-11.8.1-Source/packages/ml/src/MatrixFree/ml_MatrixFreePreconditioner.cpp
--- trilinos-11.8.1-Source.orig/packages/ml/src/MatrixFree/ml_MatrixFreePreconditioner.cpp	2014-04-30 01:14:28.000000000 +0900
+++ trilinos-11.8.1-Source/packages/ml/src/MatrixFree/ml_MatrixFreePreconditioner.cpp	2014-08-03 04:54:50.000000000 +0900
@@ -550,7 +550,7 @@
       // FIXME: use hash??
       const int& GCID = Graph.ColMap().GID(Indices[k]);
 
-      iter = find(aggregates[AID].begin(), aggregates[AID].end(), GCID);
+      iter = std::find(aggregates[AID].begin(), aggregates[AID].end(), GCID);
       if (iter == aggregates[AID].end())
         aggregates[AID].push_back(GCID);
     }
trilinos-11.8.1-Source/packages/ml/src/MatrixFreeだけに発見: ml_MatrixFreePreconditioner.cpp~
