diff -urN elpa_lib-201305.orig/CMakeLists.txt elpa_lib-201305/CMakeLists.txt
--- elpa_lib-201305.orig/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ elpa_lib-201305/CMakeLists.txt	2013-06-07 10:39:55.000000000 +0900
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 2.8)
+project(ELPA C Fortran)
+option(BUILD_SHARED_LIBS "Build shared libraries" ON)
+find_package(MPI)
+set(CMAKE_EXE_LINKER_FLAGS ${MPI_C_LINK_FLAGS})
+add_subdirectory(ELPA_2011.12)
+add_subdirectory(ELPA_2013.02.BETA)
+add_subdirectory(ELPA_development_version)
diff -urN elpa_lib-201305.orig/ELPA_2011.12/CMakeLists.txt elpa_lib-201305/ELPA_2011.12/CMakeLists.txt
--- elpa_lib-201305.orig/ELPA_2011.12/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ elpa_lib-201305/ELPA_2011.12/CMakeLists.txt	2013-06-07 10:39:55.000000000 +0900
@@ -0,0 +1,11 @@
+set(SOURCES src/elpa1.f90 src/elpa2.f90 src/elpa2_kernels.f90)
+add_library(libelpa-2011.12 ${SOURCES})
+target_link_libraries(libelpa-2011.12 ${SCALAPACK_LIB})
+install(TARGETS libelpa-2011.12 ARCHIVE DESTINATION lib LIBRARY DESTINATION lib RUNTIME DESTINATION bin)
+
+set(TARGETS read_real read_real_gen test_complex test_complex2 test_complex_gen test_real test_real2 test_real_gen)
+foreach(t ${TARGETS})
+  add_executable(${t}-2011.12 test/${t}.f90)
+  set_target_properties(${t}-2011.12 PROPERTIES OUTPUT_NAME ${t})
+  target_link_libraries(${t}-2011.12 libelpa-2011.12)
+endforeach(t ${TARGETS})
diff -urN elpa_lib-201305.orig/ELPA_2013.02.BETA/CMakeLists.txt elpa_lib-201305/ELPA_2013.02.BETA/CMakeLists.txt
--- elpa_lib-201305.orig/ELPA_2013.02.BETA/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ elpa_lib-201305/ELPA_2013.02.BETA/CMakeLists.txt	2013-06-07 10:39:55.000000000 +0900
@@ -0,0 +1,11 @@
+set(SOURCES src/elpa1.f90 src/elpa2.f90 src/elpa2_kernels/elpa2_kernels_complex.f90 src/elpa2_kernels/elpa2_kernels_real.f90)
+add_library(libelpa-2013.02.beta ${SOURCES})
+target_link_libraries(libelpa-2013.02.beta ${SCALAPACK_LIB})
+install(TARGETS libelpa-2013.02.beta ARCHIVE DESTINATION lib LIBRARY DESTINATION lib RUNTIME DESTINATION bin)
+
+set(TARGETS read_real read_real_gen test_complex test_complex2 test_complex_gen test_real test_real2 test_real_gen)
+foreach(t ${TARGETS})
+  add_executable(${t}-2013.02.beta test/${t}.f90)
+  set_target_properties(${t}-2013.02.beta PROPERTIES OUTPUT_NAME ${t})
+  target_link_libraries(${t}-2013.02.beta libelpa-2013.02.beta)
+endforeach(t ${TARGETS})
diff -urN elpa_lib-201305.orig/ELPA_2013.02.BETA/test/test_complex2.f90 elpa_lib-201305/ELPA_2013.02.BETA/test/test_complex2.f90
--- elpa_lib-201305.orig/ELPA_2013.02.BETA/test/test_complex2.f90	2013-05-21 14:44:19.000000000 +0900
+++ elpa_lib-201305/ELPA_2013.02.BETA/test/test_complex2.f90	2013-06-07 10:45:49.000000000 +0900
@@ -58,6 +58,7 @@
 
    implicit none
    include 'mpif.h'
+   integer :: iargc
 
    !-------------------------------------------------------------------------------
    ! Please set system size parameters below!
diff -urN elpa_lib-201305.orig/ELPA_2013.02.BETA/test/test_real2.f90 elpa_lib-201305/ELPA_2013.02.BETA/test/test_real2.f90
--- elpa_lib-201305.orig/ELPA_2013.02.BETA/test/test_real2.f90	2013-05-21 14:44:19.000000000 +0900
+++ elpa_lib-201305/ELPA_2013.02.BETA/test/test_real2.f90	2013-06-07 10:46:25.000000000 +0900
@@ -59,6 +59,7 @@
 
    implicit none
    include 'mpif.h'
+   integer :: iargc
 
    !-------------------------------------------------------------------------------
    ! Please set system size parameters below!
diff -urN elpa_lib-201305.orig/ELPA_development_version/CMakeLists.txt elpa_lib-201305/ELPA_development_version/CMakeLists.txt
--- elpa_lib-201305.orig/ELPA_development_version/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ elpa_lib-201305/ELPA_development_version/CMakeLists.txt	2013-06-07 10:39:55.000000000 +0900
@@ -0,0 +1,11 @@
+set(SOURCES src/elpa1.f90 src/elpa2.f90 src/elpa2_kernels/elpa2_kernels_complex.f90 src/elpa2_kernels/elpa2_kernels_real.f90 src/elpa_qr/elpa_pdgeqrf.f90 src/elpa_qr/elpa_pdlarfb.f90 src/elpa_qr/elpa_qrkernels.f90 src/elpa_qr/tum_utils.f90)
+add_library(libelpa-develop ${SOURCES})
+target_link_libraries(libelpa-develop ${SCALAPACK_LIB})
+install(TARGETS libelpa-develop ARCHIVE DESTINATION lib LIBRARY DESTINATION lib RUNTIME DESTINATION bin)
+
+set(TARGETS read_real read_real_gen test_complex test_complex2 test_complex_gen test_real test_real2 test_real_gen)
+foreach(t ${TARGETS})
+  add_executable(${t}-develop test/${t}.f90 test/read_test_parameters.f90)
+  set_target_properties(${t}-develop PROPERTIES OUTPUT_NAME ${t})
+  target_link_libraries(${t}-develop libelpa-develop)
+endforeach(t ${TARGETS})
diff -urN elpa_lib-201305.orig/ELPA_development_version/test/test_complex2.f90 elpa_lib-201305/ELPA_development_version/test/test_complex2.f90
--- elpa_lib-201305.orig/ELPA_development_version/test/test_complex2.f90	2013-05-21 14:44:19.000000000 +0900
+++ elpa_lib-201305/ELPA_development_version/test/test_complex2.f90	2013-06-07 10:46:55.000000000 +0900
@@ -58,6 +58,7 @@
 
    implicit none
    include 'mpif.h'
+   integer :: iargc
 
    !-------------------------------------------------------------------------------
    ! Please set system size parameters below!
